dslib.h

```diff
@@ -104,6 +105,30 @@ char* top(stack *top)
 		return top->d.p;
 }
 
+void emptystack(stack **top)
+{
+	if(*top == NULL)
+		return;
+		
+	while(*top != NULL) {
+		stack *dt = *top;
+		*top = (*top)->link;
+		free(dt);
+	}
+}
+
+void cleanqueue(queue **front, queue **rear)
+{
+	if (*front != NULL && *rear != NULL) {
```

Comment:
Same goes for the enqueue() and dequeue() functions. Why do you need to check if rear is NULL or not? A NULL check for front should suffice in all cases.